### PR TITLE
Sync and update of angular dependency-injection page

### PR DESCRIPTION
**Pull request by chalin**
Contributes to #737 

Staged at https://angulardart-org-dev3.firebaseapp.com/angular/guide/dependency-injection

### CODE REVIEW

File: src/angular/guide/dependency-injection.md [hunk 1]
```diff
@@ -12,26 +12,24 @@ nextpage:
...
-we really can't build an Angular app without it.
+**Dependency injection** is an important app design pattern.
 It's used so widely that almost everyone just calls it _DI_.
 
-In this chapter we'll learn what DI is and why we want it.
-Then we'll learn [how to use it](#angular-di) in an Angular app.
+Angular has its own dependency injection framework, and
+you really can't build an Angular application without it.
+
+This page covers what DI is and why it's useful.
```
Comment (kwalrath) [inline src/angular/guide/dependency-injection.md]: Adjust this intro due to having the Angular DI stuff on this page (unlike TS, I guess). Maybe:

This page covers what DI is, why it's useful, and how to use Angular DI.

File: src/angular/guide/dependency-injection.md [hunk 2]
```diff
@@ -12,26 +12,24 @@ nextpage:
...
 
-- [Why dependency injection?](#why-dependency-injection)
-- [Angular dependency injection](#angular-dependency-injection)
-- [Injector providers](#injector-providers)
-- [Dependency injection tokens](#dependency-injection-tokens)
-- [Summary](#summary)
+When you've learned the general pattern, you're ready to turn to
```
Comment (kwalrath) [inline src/angular/guide/dependency-injection.md]: Delete this paragraph.

File: src/angular/guide/dependency-injection.md [hunk 3]
```diff
@@ -352,46 +341,98 @@ Here's a revised `HeroesComponent` that registers the `HeroService`.
...
+A component-provided service may have a limited lifetime. Each new instance of the component gets its own instance of the service
+and, when the component instance is destroyed, so is that service instance.
+
+In this sample app, the `HeroComponent` is created when the application starts
+and is never destroyed so the `HeroService` created for the `HeroComponent` also lives for the life of the app.
+
+<a id="bootstrap"></a>
+### Bootstrap providers
+
+Another common common way to register a provider is using the [bootstrap()][] function.
```
Comment (kwalrath) [inline src/angular/guide/dependency-injection.md]: link isn't working

File: src/angular/guide/dependency-injection.md [hunk 4]
```diff
@@ -988,80 +1020,114 @@ constructor argument with `@Optional()`:
...
   It could acquire services from any ancestor component, not just its own.
-  We're forced to spelunk the implementation to discover what it does.
+  You're forced to spelunk the implementation to discover what it does.
 
   Framework developers may take this approach when they
   must acquire services generically and dynamically.
 </div>
+
+<a id="one-class-per-file"></a>
+## Appendix: One class per file
```
Comment (kwalrath) [inline src/angular/guide/dependency-injection.md]: Do we really need this section?

File: src/angular/guide/dependency-injection.md (hunk 2 above)
Comment (chalin) [reply src/angular/guide/dependency-injection.md]: done

File: src/angular/guide/dependency-injection.md (hunk 1 above)
Comment (chalin) [reply src/angular/guide/dependency-injection.md]: done

File: src/angular/guide/dependency-injection.md (hunk 3 above)
Comment (chalin) [reply src/angular/guide/dependency-injection.md]: fixed

File: src/angular/guide/dependency-injection.md (hunk 4 above)
Comment (chalin) [reply src/angular/guide/dependency-injection.md]: I've dropped it (and removed the link to this appendix).

**Pull request merged**